sound/soc/sof/intel/telemetry.h

```diff
@@ -0,0 +1,45 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * telemetry data in debug windows
+ */
+
+#ifndef _SOF_INTEL_TELEMETRY_H
+#define _SOF_INTEL_TELEMETRY_H
+
+#define XTENSA_TOOL_CHAIN_ZEPHYR	1
+#define XTENSA_TOOL_CHAIN_XCC		2
+#define XTENSA_SOC_INTEL_ADSP		3
+#define XTENSA_BLOCK_HDR_VER		2
+#define XTENSA_CORE_DUMP_SEPARATOR	0x0DEC0DEB
+#define XTENSA_CORE_AR_REGS_COUNT	16
+
+struct xtensa_arch_block {
+	u8 soc; /* shoud be equal to XTENSA_SOC_INTEL_ADSP */
+	u16 version;
+	u8 toolchain; /* ZEPHYR or XCC */
```

Comment:
for above 3 declarations, not sure there is alignment with sof or other driver component(both use u8 or u16).
if not, can use u32 all instead of u8 and u16? it can't save memory and have to add aligned(4) in struct declartion.

further, may be use below definition seems more reasonable:

struct xtensa_info
{
u8   soc: 8;
u16 version: 16;
u8    toolchain: 8;
}

this will give clear message to compiler that it only need 4 bytes for this structure.
just FYI

Reply:
Thanks, I see. The structure is defined in zephyr framework and SOF uses it. I want to keep align with the original style.